src/types/guild.ts

```diff
@@ -660,3 +666,40 @@ export interface EditGuildTemplate {
   /** description for the template (0-120 characters) */
   description?: string | null;
 }
+
+/** */
+export interface MembershipScreeningPayload {
+  /** when the fields were last updated */
+  version: string;
+  /** the steps in the screening form */
+  form_fields: MembershipScreeningFieldPayload[];
+  /** the server description shown in the screening form */
+  description: string | null;
+}
+
+/** */
+export interface MembershipScreeningFieldPayload {
+  /** the type of field */
+  field_type: MembershipScreeningFieldTypes;
+  /** the title of the field */
+  label: string;
+  /** the list of rules */
+  values?: string[];
+  /** whether the user has to fill out this field */
+  required: boolean;
+}
+
+/** */
+export type MembershipScreeningFieldTypes =
+  /** Server Rules */
+  "TERMS";
+
+/**  */
+export interface ModifyGuildMembershipScreeningFormParams {
+  /** whether Membership Screening is enabled */
+  enabled?: boolean;
+  /** array of field objects serialized in a string */
+  form_fields?: string;
```

Comment:
This is a user end interface? This should not use snake case

Reply:
This is an API type―parameters for the "modify guild membership screening form" endpoint.

Reply:
I believe this interface is also used for `options?: ` for the user to provide options

Reply:
@Skillz4Killz, you are right, hmm... I should change that.